### ISSUE TITLE
Implement consistent snapshot feature

### ISSUE THREAD

**Issue by heartsucker** [heartsucker/rust-tuf#96]
Depends on #95 

This would be a nice-to-have for the rustup milestone, but I'm leaving it off for now since that milestone is really shooting for the MVP that would work behind a CLI feature flag (at least initially).

**Comment by heartsucker** [comment]
This would actually mean undoing some of 04993c5 because the hashes are needed to identify the "absolute" path to a piece of metadata when addressing them in consistent snapshot mode.

**Comment by heartsucker** [comment]
In the conversation with the Rust team, it looks like `static.rust-lang.org` uses both S3 and Cloudfront which means there are the problems of eventual consistency and caching respectively. For this to work in `rustup`, this feature is a hard requirement.
